### PR TITLE
move span, trace and event implementation to sdk.

### LINKED ISSUE #20: Separate API from SDK in Go repo
The current prototype that was merged into this repository was unfinished work, in part because it was created while the OpenTelemetry specification was not finished. 

The prototype included several concepts at the API level that were dedicated to streaming events to an asynchronous or out-of-process observer as part of the PoC. These types, in particular "ScopeID" are not needed for an in-process span and metrics exporter. To complete this work, the API should have a pure interface that an SDK can provide, for example:

https://github.com/tigrannajaryan/otelapi-compat/

### PR DESCRIPTION
**Pull request by rghetia**
partially fixes #20.

### CODE REVIEW

File: api/trace/noop_span.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+	"context"
+
+	"google.golang.org/grpc/codes"
+
+	"github.com/open-telemetry/opentelemetry-go/api/core"
+	"github.com/open-telemetry/opentelemetry-go/api/event"
+)
+
+// SpancContext returns an invalid span context.
+func (sp *noopSpan) SpanContext() core.SpanContext {
```
Comment (freeformz) [inline api/trace/noop_span.go]: Suggestion if this type is expected to conform to an interface declare that here via:

```go
var _ <Interface> = &noopSpan{}
```

File: plugin/httptrace/clienttrace.go [hunk 2]
```diff
@@ -46,6 +49,9 @@ var (
 	HTTPHeaderMIME = tag.New("http.mime")
 	HTTPRemoteAddr = tag.New("http.remote")
 	HTTPLocalAddr  = tag.New("http.local")
+	messageKey     = tag.New("message",
```
Comment (sjkaris) [inline plugin/httptrace/clienttrace.go]: Should this also be exported as the other tags are?

File: sdk/trace/trace.go [hunk 3]
```diff
@@ -18,15 +18,16 @@ import (
...
 
 	"google.golang.org/grpc/codes"
 
 	"github.com/open-telemetry/opentelemetry-go/api/core"
-	"github.com/open-telemetry/opentelemetry-go/api/event"
 	"github.com/open-telemetry/opentelemetry-go/api/scope"
 	"github.com/open-telemetry/opentelemetry-go/api/tag"
+	apitrace "github.com/open-telemetry/opentelemetry-go/api/trace"
 	"github.com/open-telemetry/opentelemetry-go/exporter/observer"
+
```
Comment (sjkaris) [inline sdk/trace/trace.go]: nit: no newline here

File: plugin/httptrace/clienttrace.go [hunk 4]
```diff
@@ -22,9 +22,12 @@ import (
 	"strings"
 
 	"github.com/open-telemetry/opentelemetry-go/api/core"
-	"github.com/open-telemetry/opentelemetry-go/api/event"
 	"github.com/open-telemetry/opentelemetry-go/api/tag"
 	"github.com/open-telemetry/opentelemetry-go/api/trace"
+
+	"google.golang.org/grpc/codes"
+
+	"github.com/open-telemetry/opentelemetry-go/sdk/event"
```
Comment (sjkaris) [inline plugin/httptrace/clienttrace.go]: nit: should be moved up

File: example/http/client/client.go [hunk 5]
```diff
@@ -21,16 +21,17 @@ import (
...
 
 	"github.com/open-telemetry/opentelemetry-go/api/tag"
-	"github.com/open-telemetry/opentelemetry-go/api/trace"
+	apitrace "github.com/open-telemetry/opentelemetry-go/api/trace"
 	"github.com/open-telemetry/opentelemetry-go/plugin/httptrace"
 
 	"google.golang.org/grpc/codes"
 
 	_ "github.com/open-telemetry/opentelemetry-go/exporter/loader"
+	"github.com/open-telemetry/opentelemetry-go/sdk/trace"
```
Comment (sjkaris) [inline example/http/client/client.go]: `google.golang.org/grpc/codes` should go up a section and the `github.com/open-telemetry/opentelemetry-go` sections should be joined

File: api/trace/noop_span.go (hunk 1 above)
Comment (rghetia) [reply api/trace/noop_span.go]: thx. Done.

File: example/http/client/client.go (hunk 5 above)
Comment (rghetia) [reply example/http/client/client.go]: done.

File: plugin/httptrace/clienttrace.go (hunk 4 above)
Comment (rghetia) [reply plugin/httptrace/clienttrace.go]: done.

File: plugin/httptrace/clienttrace.go (hunk 2 above)
Comment (rghetia) [reply plugin/httptrace/clienttrace.go]: I have exported it for now but I wonder if other should be exported.

File: sdk/trace/trace.go (hunk 3 above)
Comment (rghetia) [reply sdk/trace/trace.go]: removed.

**Pull request merged**